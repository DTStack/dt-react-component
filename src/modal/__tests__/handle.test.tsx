import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Handler from '../handle';

describe('Handler Component', () => {
    it('should match snapshot', () => {
        const { asFragment } = render(<Handler data-testid="handler" handleAxis="x" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should forward ref to the div element', () => {
        const ref = React.createRef<HTMLDivElement>();
        render(<Handler handleAxis="y" ref={ref} />);
        expect(ref.current).toBeInstanceOf(HTMLDivElement);
        expect(ref.current).toHaveClass('dt-modal-resize-handle handle-y');
    });
});
