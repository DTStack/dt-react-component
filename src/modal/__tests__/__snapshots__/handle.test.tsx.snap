// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handler Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="dt-modal-resize-handle handle-x"
    data-testid="handler"
  />
</DocumentFragment>
`;
